### ISSUE TITLE
Facilitator role for async UX

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/action#2918]
If a user starts a meeting with the intent of opening a long-running, async meeting UX (e.g. starting a Retro when a new sprint starts, intending to conclude about 2-3 weeks later)…

- How do we persist this person as the facilitator? Is that a hard rule?
- Can the role be assumed seamlessly by others on the team? How to pass the role to unblock the meeting if the facilitator can’t continue in the role?
- How do we know the async intent without an explicit async meeting type? By phase timebox start?
- We plan for the timebox affordance to notify the team on timebox start and the facilitator on timebox end. If the role gets assumed by another user, does the new facilitator get the notification?

### Acceptance Criteria
- Proposed candidate solutions for how to persist facilitator as comments below
- Decided how the facilitator role switching logic is updated in meeting model
- Wrote any necessary implementation or design issue(s)

### Estimated Effort
- 11 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by jordanh** [comment]
Could this be as simple as:
   - If the meeting has ever had a timebox applied, we assume it's async and do not rotate the facilitator role

What do we think?

**Comment by ackernaut** [comment]
>If the meeting has ever had a timebox applied, we assume it's async and do not rotate the facilitator role

I think that is probably a good place to start. We should try it. A few notes:

Currently, folks can assume the facilitator role without having to wait for an approval. I think this is a good thing, allowing folks more flexibility. However, it’s not immediately clear how to take over. This is okay to a degree because this is not a primary action in the meeting views. I do think we could explore an IA solution to make switching facilitator easier.

After that we could consider defaulting to manual selection of the facilitator. I’m not sure which case is more frequent, for a facilitator to drop connection briefly, then come back and want the controls again. Or for somebody to drop and the team decides to move on without them. I would guess the first is more common and having to play a game of tag isn’t probably the best UX. If there is a longer play at hand, well folks can always assign a new facilitator manually. I think the automatic switching is a little premature given the UX case assumption.

Describing the IA with words:

- The left drawer menu is about the meeting meta, navigation, and global help. There’s a new control that shows who’s the facilitator (aka who’s guiding the group through the meeting phases). Tapping the control reveals the menu of teammates (like a task card) and a new facilitator can be chosen.
- The top bar has the phase name, an iconic toggle for the left menu when closed, and an iconic toggle for the right menu with closed. We probably move the contextual help menu to a small ? icon toggle here because of proximity and to clean up the bottom bar area.
- The right drawer menu is the team hub. There’s a list of teammates (i.e. all possible participants) whether connected or not and any related meta (e.g. time zone). It also has a tab to the team discussion that delineates along with the meeting phases (deeper details at another time). Here one can quickly manage invites (share a link, invite to team, etc.)
- The main area contains any prompts and hints (no longer are they in the top bar), along with the card views
- The bottom area is where I control my presence (including a/v connection, screen sharing) and my contextual input (e.g. add an agenda item, task card, reflection card, etc.). This floats at the bottom of the viewport, or above the bottom bar when facilitating.
- The bottom bar is currently reserved for the facilitator. The main purpose is meeting navigation and controls.
- The a/v module is probably only 1 active avatar, who’s speaking at the time, with a control to switch to gallery etc. This module can be dragged and pins to the sides. When there’s enough room in the viewport I’ve thought about showing maybe up to 3-5 folks based on activity etc. But only showing 1, the active speaker, may afford a cleaner, minimal UX at all views.

Updated IA drawings and concepts in progress…

Here I think we do 2 things:
1) Write an implementation issue that locks in a faciliator during a timebox (date/time), 10.5 pts
2) Update IA concepts to show a clearer affordance for selecting the facilitator, 10.5 pts, and could be a contribution toward #2926

**Comment by ackernaut** [comment]
@mattkrick @jordanh 

### TL;DR

What is the most likely case? A facilitator drops briefly and expects to still be facilitator when they get back in (while juggling with others “who’s the facilitator?”). Or a team just moves on as they wait for the person who was facilitating to come back. The first case could be more of a confusing UX, esp. with early usage of the project. On the other hand, the team that is used to moving along would probably learn how to pass the baton (using the existing affordance, avatar menus, or a new affordance)

- Do we disable automatic facilitator role switching only during time box?
- Do we disable it all the time except in the case that a team member is removed or leaves? This would require folks to select a new facilitator, which may reduce some confusion
- Does both the person who started the timebox, and the current facilitator (if the facilitator role changed hands), both get the Slack notif when it ends? or only the current facilitator?
- Do we add [an explicit facilitator control](https://www.figma.com/file/0lOTlvsj87GBFFzgee9OkVWi?node-id=35%3A0&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#7813453) to the meeting navigation? This decouples the information and affordance from a/v avatars, team menu, and stage navigation (all of which are a proposed target IA)

**Comment by mattkrick** [comment]
- a meetng has a `facilitator` and a `defaultFacilitator`
- when a meeting is started, both are set to the person who started the meeting
- when a facilitator disconnects, another member elects themselves as the `facilitator`
- when someone reconnects, if they are the `defaultFacilitator`, they get promoted to `facilitator`
- when someone promotes themselves to facilitator (not due to disconnect) they become the `facilitator` and `defaultFacilitator`
- when a timebox is active and there is no connected facilitator, no other members automatically attempt to promote themselves

**Comment by jordanh** [comment]
> What is the most likely case? A facilitator drops briefly and expects to still be facilitator when they get back in (while juggling with others “who’s the facilitator?”).

When a timebox is active, this is the most likely case. @mattkrick's suggested AC seem correct to me:

> - a meetng has a `facilitator` and a `defaultFacilitator`
> - when a meeting is started, both are set to the person who started the meeting
> - when a facilitator disconnects, another member elects themselves as the `facilitator` unless a timebox is active
> - when someone reconnects, if they are the `defaultFacilitator`, they get promoted to `facilitator` 
> - when someone promotes themselves to facilitator (not due to disconnect) they become the `facilitator` and `defaultFacilitator`

I will open an implementation issue now.
